### DISCUSSION TITLE
Ancient woodlands

### DISCUSSION THREAD

**Discussion by colmjude** [digital-land/data-standards-backlog#32] [Planning consideration]
# Ancient woodlands

**Comment by greg-slater** [comment]
Natural England have published a [new dataset](https://services.arcgis.com/JJzESW51TqeY9uat/arcgis/rest/services/Ancient_Woodland_Revised_England/FeatureServer/2/query?outFields=*&where=1%3D1) created using a completely new methodology. This dataset takes precedence over the [previous dataset](https://naturalengland-defra.opendata.arcgis.com/datasets/Defra::ancient-woodland-england/about), and the new records use a new, persistent, unique identifier. However, while the previous dataset had national coverage, the new dataset has only been published for a few counties so far, and they plan to release more data county-by-county through the next year. Some questions which need answering before we can add this new data:

* should we completely end-dateall of the existing entities and cutover to the new dataset even though it only currently has partial coverage?
* or should we aim to migrate area by area as Natural England publish updates? This method would rely on knowing for certain which geographies natural england are using as areas I think, so we can reliably end-date the correct existing entities within the same area.

**Reply by stevenjmesser** [reply]
What are the pros and cons of each approach? Consider discussing these as a team, considering the impact on end users and data integrity.

**Reply by greg-slater** [reply]
Adding some notes on possible approaches having done some initial analysis of the two datasets.

## Possible approaches

### 1. retire all of existing data, add new partial dataset

Pros
+ Simple to implement, it's easy to end-date all entities from a single endpoint/resource
+ Simple to maintain, we just run the assign-entities process when new data is added

Cons
- Bad for users, existing ancient-woodland data would vanish for a lot of the country

### 2. retire existing data only for areas where new data has been published

Pros
+ Good for users, full national coverage maintained

Cons
- Possibly not clear to users that data in different areas has come from different sources and produced by different methodologies
- Manual adding: we'd need to identify the existing entities to retire based on the area new data is supplied for
- We'd need to end-date individual entities, currently this requires self-hosting an additional dataset and adding it as an endpoint, and this process may change in the future
- Manual maintenance: each time data is added we'd need to identify the next batch of existing entities to retire. We could create a re-runnable script for this, but there would be some element of uncertainty if we're not completely certain which geographies Natural England are working to

### 3. add new release to a new dataset, and keep both releases live alongside each other until new dataset is nationally complete

Pros
+ Possibly clearer to users that there are differences between the datasets
+ Simple to add and maintain, don't need to worry about retiring individual entities

Cons
- Approach unlikely to fit with intended data model
- Possibly confusing to users to have multiple different ancient woodland datasets

---- 

Possible process for approach no. 2:

1. Add new release as new endpoint and assign new entities
2. Build ancient-woodland collection locally (or push config changes to staging and build there)
3. Run script to identify entities with a start date before the new endpoint was added that are in the areas new data has been supplied for
4. Create end-date dataset for existing ancient-woodland entities to be retired
5. Add end-date dataset as endpoint
6. Re-build collection to test end-date changes
7. Push live

Each time new data is published (identified by unknown entity monitoring):
1. Assign new ancient-woodland entities
2. Build ancient-woodland collection locally (or push config changes to staging and build there)
3. Run script to identify entities with a start date before the new endpoint was added that are in the areas new data has been supplied for.
4. Add new entries to end-date dataset
